libraries/3rdparty/CMakeLists.txt

```diff
@@ -46,3 +46,37 @@ set_target_properties(tinycbor PROPERTIES
 target_include_directories(tinycbor PUBLIC
     "${ROOT_DIR}/libraries/3rdparty/tinycbor/src"
 )
+
+file(GLOB MBEDTLS_FILES CONFIGURE_DEPENDS "${ROOT_DIR}/libraries/3rdparty/mbedtls/library/*.c")
+
+add_library(mbedtls
```

Comment:
You could probably optimize this further and do something similar to the jobs demo by calling find_library: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/5015cdf04561ce6e9d4e25a3726fe71dc17cdd0a/demos/jobs/jobs_demo_mosquitto/CMakeLists.txt#L18 so that mbedtls is only compiled when it doesn't already exist on the system. I'm unsure if all our code is compatible with the latest mbedtls 3.0.0, though.

Reply:
Yeah, newer version didnt compile for me. Also we use some non-default features, so system lib would not likely have them.